src/Authentication/Authenticators/JWT.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+namespace CodeIgniter\Shield\Authentication\Authenticators;
+
+use CodeIgniter\HTTP\IncomingRequest;
+use CodeIgniter\I18n\Time;
+use CodeIgniter\Shield\Authentication\AuthenticationException;
+use CodeIgniter\Shield\Authentication\AuthenticatorInterface;
+use CodeIgniter\Shield\Authentication\Authenticators\JWT\Firebase;
+use CodeIgniter\Shield\Authentication\Authenticators\JWT\JWTDecoderInterface;
+use CodeIgniter\Shield\Entities\User;
+use CodeIgniter\Shield\Exceptions\RuntimeException;
+use CodeIgniter\Shield\Models\TokenLoginModel;
+use CodeIgniter\Shield\Models\UserModel;
+use CodeIgniter\Shield\Result;
+use InvalidArgumentException;
+use stdClass;
+
+/**
+ * Stateless JWT Authenticator
+ */
+class JWT implements AuthenticatorInterface
+{
+    /**
+     * @var string Special ID Type.
+     *             This Authenticator is stateless, so no `auth_identities` record.
+     */
+    public const ID_TYPE_JWT = 'jwt';
```

Comment:
Will this ever change? With only one possible value doesn't seem necessary to use a constant.

Reply:
> Will this ever change?

It might change. `jwt-jws` may be accurate.
Or just `jws` is better?

It is used twice in the file, and used twice in `JWTAuthenticatorTest`.
If we use string `jwt`, IDE can't find where it is used easily.

> With only one possible value doesn't seem necessary to use a constant.

For example, if a dev want to change the value, it is easier when a constant is used,
even if it is only one possible value. 

And another value would be added in the future.